src/AttestationData.ts

```diff
@@ -0,0 +1,408 @@
+type TContextField = string | {type: string; data: string}
+
+///////////////////////////////////////////////////
+// Base attestation dataStr type
+///////////////////////////////////////////////////
+export interface IBaseAtt {
+  '@context': TContextField | Array<TContextField>
+  // Date/time attestation was performed
+  date?: string
+
+  // Secondary provider
+  provider?: string
+
+  // Unique ID for attestation subject from provider
+  id?: number | string
+
+  // Date/time when attestation should be considered void (e.g., for credential expiry)
+  expiry_date?: string
+
+  // Optional summary object (often useful for multiple-account types - otherwise, the data field is preferable for most of these fields, for simplicity's sake
+  summary?: {
+    // Single date/time during which attestation was applicable
+    date?: TDate
+
+    // Start date/time of period during which attestation was applicable
+    start_date?: TDate
+
+    // End date/time of period during which attestation was applicable
+    end_date?: TDate
+
+    // Different levels of generality - zero-indexed, with increasing numbers indicating less generality, to allow for unlimited levels of depth (in practice, 3-5 should be sufficient for most cases).  This allows for arbitrary levels or amounts of generality within sub-attestations, to promote an attestation subject's ability to partially disclose the amount of data provided in an attestation.
+    generality?: number
+
+    // ...extensible with other fields that summarize the content of the attestation - e.g., a list of addresses, accounts, totals of statistics, etc.
+  }
+
+  // Core attestation data, dependent on attestation type
+  data?: Array<Object | string | number> | Object | string | number
```

Comment:
If we know the supported types for data as seen below and they all look like `TSomeType | Array<TSomeType>`, could we make `IBaseAtt` generic via `IBaseAtt<TAttestation>` and then have data be of type `data?: Array<TAttestation> | TAttestation` where `TAttestation` is `TSomeType`?

Reply:
There will be more than one exception (currently I think just DOB).  I'm considering using string types inside the top-level attestation specification to force an attestation type's adherence to a given structure.